doc/code_data_relocation/relocation.rst

```diff
@@ -0,0 +1,97 @@
+.. _code_data_relocation:
+
+CODE DATA RELOCATION
+####################
+
+Overview
+********
+This script will relocate .text, .rodata, .data and .bss sections from
+required files and places it in the required memory region. This
+memory region and file are given to this python script in the form
+of a string. This script is always invoked from inside cmake.
+
+This script provides a robust way to re-order
+the memory contents without actually having to modify the code.
+In simple terms this script will do the job of
+``__attribute__((section("name")))`` for a bunch of files together.
+
+Details
+*******
+The memory region and file are given to this python script in the form of a string.
+
+Example of such a string would be:
+``SRAM2:/home/xyz/zephyr/samples/hello_world/src/main.c,SRAM1:/home/xyz/zephyr/samples/hello_world/src/main2.c``
+
+This script is invoked with following parameters:
+``python3 gen_relocate_app.py -i input_string -o generated_linker -c generated_code``
+
+Kconfig :option:`CONFIG_CODE_DATA_RELOCATION` option,  when enabled in ``prj.conf``, will
+invoke the script and do the required relocation.
+
+This script also trigger the generation of ``linker_relocate.ld`` and ``code_relocation.c`` files.
+The ``linker_relocate.ld`` file creates appropriate sections and
+link the required functions or variables from all the selected files.
+
+**Note**:
```

Comment:
This would present better as:
```
.. note:: After running this script, the text section is split into 2 parts.
   The first section will have some info regarding vector tables and debug info.
   The second section will have the complete text section.
   This is needed to force the required functions and data variables to the correct locations.
   This is due to the behavior of the linker. The linker will only link once and
   hence this text section had to be split to make room for the generated linker script.
```